test/TestProgramSettings.cc

```diff
@@ -214,4 +221,82 @@ TEST_F(ProgramSettingsTest, MultipleImagesFromPositional) {
     EXPECT_EQ(settings.files[0], casa_image.string().substr(1));
     EXPECT_EQ(settings.files[1], fits_image.string().substr(1));
     EXPECT_EQ(settings.files[2], hdf5_image.string().substr(1));
-}
\ No newline at end of file
+}
+
+TEST_F(ProgramSettingsTest, ExpectedValuesLongJSON) {
+    auto json_string = R"(
+    {
+        "verbosity": 6,
+        "no_log": true,
+        "no_http": true,
+        "no_browser": true,
+        "host": "helloworld",
+        "port": 1234,
+        "grpc_port": 5678,
+        "omp_threads": 10,
+        "top_level_folder": "/tmp",
+        "frontend_folder": "/var",
+        "exit_timeout": 10,
+        "initial_timeout": 11
+    })";
+    nlohmann::json j = nlohmann::json::parse(json_string);
+
+    carta::ProgramSettings settings;
+    settings.SetSettingsFromJSON(j);
+
+    EXPECT_EQ(settings.verbosity, 6);
+    EXPECT_EQ(settings.no_log, true);
+    EXPECT_EQ(settings.no_http, true);
+    EXPECT_EQ(settings.no_browser, true);
+    EXPECT_EQ(settings.host, "helloworld");
+    EXPECT_EQ(settings.port, 1234);
+    EXPECT_EQ(settings.grpc_port, 5678);
+    EXPECT_EQ(settings.omp_thread_count, 10);
+    EXPECT_EQ(settings.top_level_folder, "/tmp");
+    EXPECT_EQ(settings.frontend_folder, "/var");
+    EXPECT_EQ(settings.wait_time, 10);
+    EXPECT_EQ(settings.init_wait_time, 11);
+}
+
+TEST_F(ProgramSettingsTest, ValidateJSONFromFile) {
+    const std::string input = "./data/settings-good-fields.json";
+
+    carta::ProgramSettings settings;
+    auto j = settings.JSONSettingsFromFile(input);
+    settings.SetSettingsFromJSON(j);
```

Comment:
Because we're testing `JSONSettingsFromFile` in these tests, maybe we should be checking the JSON object output by that function, rather than the final settings object. There could be a separate test to make sure that `SetSettingsFromJSON` correctly leaves some or all values set to default if provided with a JSON object which is empty or has some entries missing. Right now these tests are end-to-end tests of both functions together.